packages/shared/src/modules/users/libs/validation-schemas/user-sign-up.validation-schema.ts

```diff
@@ -28,7 +43,37 @@ const userSignUp: z.ZodType<UserSignUpRequestDto> = z
 			.trim()
 			.min(UserValidationRule.PASSWORD_MINIMUM_LENGTH, {
 				message: UserValidationMessage.PASSWORD_MINIMUM_LENGTH,
-			}),
+			})
+			.max(UserValidationRule.PASSWORD_MAXIMUM_LENGTH, {
+				message: UserValidationMessage.PASSWORD_TOO_LONG,
+			})
+			.refine(
```

Comment:
let's use regex method for all regexes
```suggestion
			.regex(
```